src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -422,9 +422,29 @@ internal Accessibility LocalAccessibility
         /// <summary>
         /// Indicates whether this accessor is readonly due to reasons scoped to itself and its containing property.
         /// </summary>
-        internal override bool IsDeclaredReadOnly => LocalDeclaredReadOnly || _property.HasReadOnlyModifier || IsReadOnlyAutoGetter;
+        internal override bool IsDeclaredReadOnly
+        {
+            get
+            {
+                if (LocalDeclaredReadOnly || _property.HasReadOnlyModifier)
+                {
+                    return true;
+                }
 
-        private bool IsReadOnlyAutoGetter => ContainingType.IsStructType() && !_property.IsStatic && _isAutoPropertyAccessor && MethodKind == MethodKind.PropertyGet;
+                // We can't emit the synthesized attribute for netmodules, so in this case we consider auto-getters **not** implicitly readonly.
+                bool isBadNetModule = DeclaringCompilation.Options.OutputKind == OutputKind.NetModule &&
```

Comment:
>bool isBadNetModule = [](start = 16, length = 21)

Minor point: This might be easier to read as:
```
if (DeclaringCompilation.Options.OutputKind == NetModule &&
    DeclaringCompilation.GetWellKnownType(...).IsErrorType)
{
    return false;
}

return ContainingType.IsStructType() && ... ;
```

Reply:
Done